aws/resource_aws_budgets_budget.go

```diff
@@ -226,9 +331,77 @@ func resourceAwsBudgetsBudgetRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("time_unit", budget.TimeUnit)
 
+	return resourceAwsBudgetsBudgetNotificationRead(d, meta)
+}
+
+func resourceAwsBudgetsBudgetNotificationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+
+	describeNotificationsForBudgetOutput, err := client.DescribeNotificationsForBudget(&budgets.DescribeNotificationsForBudgetInput{
+		BudgetName: aws.String(budgetName),
+		AccountId:  aws.String(accountID),
+	})
+
+	if err != nil {
+		return err
```

Comment:
We should return error context here for operators and code maintainers:

```suggestion
		return fmt.Errorf("error describing Budget (%s) Notifications: %s", d.Id(), err)
```